src/app/clusters/ota-requestor/DefaultOTARequestor.cpp

```diff
@@ -638,7 +638,6 @@ void DefaultOTARequestor::OnDownloadStateChanged(OTADownloader::State state, OTA
     {
     case OTADownloader::State::kComplete:
         mOtaRequestorDriver->UpdateDownloaded();
-        mBdxMessenger.Reset();
```

Comment:
In addition, you should be calling `Abort` on the EC and not `Close` in the cases where you do have a valid `mpExchangeCtx` present. If you're at a stage where you're terminating the EC, I don't believe you want to actually keep it around in the re-transmission table since the intention is to completely shut-down all interactions at that point.